tokio-io/src/lib.rs

```diff
@@ -53,7 +53,7 @@ mod framed;
 mod framed_read;
 mod framed_write;
 mod length_delimited;
-mod lines;
+pub mod lines;
```

Comment:
This is needed as `tokio::io` needs to `pub use` it, so the temp docs in trunk below is needed.

What should we do in this case?

Reply:
Since `tokio-io` exports `lines`/`Lines` publicly in the `tokio_io::io` module (although the _code_ lives in the private `tokio_io::lines` module), and we're moving everything in `tokio_io::io`, I think we ought to copy the `lines` module into `tokio::io` as well. Then, `tokio::io` can just `pub use self::lines::{lines, Lines}`, and we don't have to make this`pub`.

We should be able to deprecate the `lines` exported in the old location after doing so.